common/system_power.c

```diff
@@ -22,3 +26,33 @@ system_reset(void)
 	}
 	panic("Failed to reset system");
 }
+
+void
+system_shutdown(void)
+{
+	struct device *pmic, *regl;
+
+	prcm_shutdown();
+
+	/* Disable voltage regulators for each regulator driver. */
+	regl = dm_first_dev_by_class(DM_CLASS_REGULATOR);
```

Comment:
```
struct device *regl;
uint8_t id;

for (regl = dm_get_subdev_by_index(DM_CLASS_REGULATOR, 0, &id;
     regl != NULL; regl = dm_next_subdev(regl, &id)) {
        regulator_disable(regl, id);
}
```

Reply:
That is _substantially_ less readable, but I will change it.

Reply:
Done.